### PR TITLE
[Auth] Org scope

### PR DESCRIPTION
**Pull request by sashankg**
- **feat(oauth): add permission scope types, parsing, and matching**
- **fixup: remove repo from validResources, use Permission structs directly in tests**
- **feat(oauth): enforce admin-only org scope grant and custom scope validation**
- **fixup: address code review - use strings.HasPrefix, add admin gate test, fix duplicate require**
- **fixup: pass nil auth to NewServer in jetstream tests**
- **fixup: add strings.HasPrefix guard in HandleCallback, fix TokenValidator return type to syntax.DID**
- **fixes**

### CODE REVIEW

File: internal/jetstream/server.go [hunk 1]
```diff
@@ -8,41 +8,42 @@ import (
...
-// This will be added as a follow up.
-
-// The Server handles listening for updates from an ingestion stream and fanning them out to many upstream
-// jetstream subscribers over HTTP SSE. We don't use Websocket because it's more expensive and complex operationally,
-// and this usecase doesn't support the client sending messages.
-//
-// The official jetstream API uses bi-directionality so that the client can update
-// what collections / repos / etc. it is subscribed to. This is a simple enough usecase
-// that the client can close the connection and re-open it with the new parameters and the
-// timestamp cursor from where it left off.
```
Comment (arushibandi) [inline internal/jetstream/server.go:28]: nooo don't delete my comments!!

File: internal/oauthserver/permission.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
+package oauthserver
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+var validResources = map[string]bool{
+	"org": true,
```
Comment (arushibandi) [inline internal/oauthserver/permission.go]: nit: typed constants always

File: internal/oauthserver/permission.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+type permission struct {
+	Resource   string
+	Collection string
```
Comment (arushibandi) [inline internal/oauthserver/permission.go]: can we make this `syntax.NSID`?

File: internal/oauthserver/permission.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+	Actions    []string
```
Comment (arushibandi) [inline internal/oauthserver/permission.go]: same thing typed constants here? `[]permissionAction`

File: internal/oauthserver/permission.go [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+	"net/url"
+	"strings"
+)
+
+var validResources = map[string]bool{
+	"org": true,
+}
+
+type permission struct {
+	Resource   string
```
Comment (arushibandi) [inline internal/oauthserver/permission.go]: same thing

File: internal/oauthserver/permission.go [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+	if granted.Collection != "*" && granted.Collection != required.Collection {
+		return false
+	}
+	if len(required.Actions) == 0 {
+		return true
+	}
+	if len(granted.Actions) == 0 {
+		return true
+	}
+	requiredSet := make(map[string]bool, len(required.Actions))
```
Comment (arushibandi) [inline internal/oauthserver/permission.go]: you can do `xmaps.Set` / make it `map[string]struct{}` for a set (so there's no true | false | existence triple)

File: internal/oauthserver/permission.go [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+	for _, a := range required.Actions {
+		requiredSet[a] = true
+	}
+	for _, a := range granted.Actions {
+		delete(requiredSet, a)
```
Comment (arushibandi) [inline internal/oauthserver/permission.go]: there might be some kind of `maps.` helper to use here even

File: internal/oauthserver/permission.go [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+	}
+	return len(requiredSet) == 0
+}
+
+func scopesSatisfy(grantedScopes, requiredScopes []string) bool {
```
Comment (arushibandi) [inline internal/oauthserver/permission.go]: i think i'm blind but where does this get used?

**Review by arushibandi — APPROVED** [review]
